### PR TITLE
dependencies: upgrading to v0.17.0 of github.com/hashicorp/go-azure-helpers

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR:

* Updates to v0.17.0 of `github.com/hashicorp/go-azure-helpers`
* Updates to v2.8.0 of `github.com/hashicorp/terraform-plugin-sdk`
* Updates the `location` package to use the upstream `commonschema` common location schema

v0.17.0 of GoAzureHelpers intentionally moves some files around so there is a breaking change in here which is why this is pretty large - however I've split this into commits, so commit-by-commit this should be fairly clear.

### CODE REVIEW

**Review by koikonom — APPROVED** [review]
LGTM

File: internal/location/schema.go [hunk 1]
```diff
@@ -1,62 +1,38 @@
...
 func Schema() *pluginsdk.Schema {
-	return &pluginsdk.Schema{
-		Type:             pluginsdk.TypeString,
-		Required:         true,
-		ForceNew:         true,
-		ValidateFunc:     EnhancedValidate,
-		StateFunc:        StateFunc,
-		DiffSuppressFunc: DiffSuppressFunc,
-	}
+	return commonschema.LocationSchema()
```
Comment (katbyte) [inline internal/location/schema.go:10]: does it make sense to have schema on the function name? other functions in there don't?

File: internal/location/schema.go [hunk 2]
```diff
@@ -1,62 +1,38 @@
...
-// where these are Required and can be changed
 func SchemaWithoutForceNew() *pluginsdk.Schema {
-	return &pluginsdk.Schema{
-		Type:             pluginsdk.TypeString,
-		Required:         true,
-		ValidateFunc:     EnhancedValidate,
-		StateFunc:        StateFunc,
-		DiffSuppressFunc: DiffSuppressFunc,
-	}
+	return commonschema.LocationSchemaWithoutForceNew()
```
Comment (katbyte) [inline internal/location/schema.go:22]: and here

**Review by katbyte — CHANGES_REQUESTED** [review]
The schema on some functions and not others i called out in my late review on https://github.com/hashicorp/go-azure-helpers/pull/85 - as well as my concerns with how the pointer functions wording is confusing to read

**Review by katbyte — APPROVED** [review]
As discussed approving for now - will address comments in a future pr

**Comment by tombuildsstuff** [comment]
Sanity check tests pass:

Screenshot 2021-11-05 at 19 01 59

File: internal/location/schema.go (hunk 1 above)
Comment (tombuildsstuff) [reply internal/location/schema.go:10]: (as discussed) will pick this off OOB in the next helpers release

**Pull request merged**